signac/core/collection_redis.py

```diff
@@ -18,33 +19,22 @@ class RedisCollection(SyncedCollection):
 
     backend = __name__  # type: ignore
 
-    def __init__(self, name=None, client=None, redis_kwargs=None, **kwargs):
-        import redis
-
-        if client is None:
-            redis_kwargs = redis_kwargs if redis_kwargs is not None else {}
-            self._client = redis.Redis(**redis_kwargs)
-        else:
-            self._client = client
-        self._name = name
+    def __init__(self, client=None, **kwargs):
+        self._client = client
         super().__init__(**kwargs)
-        if (name is None) == (self._parent is None):
-            raise ValueError(
-                "Illegal argument combination, one of the two arguments, "
-                "parent or name must be None, but not both.")
 
     def _load(self):
         """Load the data from a Redis-database."""
         blob = self._client.get(self._name)
-        return json.loads(blob) if blob is not None else blob
+        return None if blob is None else json.loads(blob)
 
     def _sync(self):
         """Write the data from Redis-database."""
-        data = self.to_base()
-        # Serialize data:
-        blob = json.dumps(data).encode()
+        self._client.set(self._name, json.dumps(self.to_base()).encode())
 
-        self._client.set(self._name, blob)
+    def __deepcopy__(self, memo):
+        return type(self)(client=self._client, name=self._name, data=self.to_base(),
```

Comment:
For a proper deepcopy you have to deepcopy `self._client` and `self._name` as well.

Reply:
I tried to do that but `deepcopy(self._client)` raise `TypeError`.

Reply:
In cases like this, do not silently **introduce a bug into the code**, but instead point out the problem in the discussion here and propose an alternative solution if you have any. I anticipated this as a potential issue, but I did not want to make any assumptions before we actually test this out.

To address this issue, I suggest that we rename `__deepcopy__` to `_pseudo_deepcopy` and use that function in the tests instead. In this way we ensure that tests are as closely similar as possible, without implementing a flawed `__deepcopy__()` function. So in the test, it should look like this:
```python
try:
    synced_dict2 = deepcopy(synced_dict)
except TypeError:
    # Use fallback implementation, deepcopy not supported by backend.
    synced_dict2 = synced_dict._pseudo_deepcopy()
```
Does this make sense? If not, please ask.

Reply:
Yes, it makes sense.